utilities/include/vec4.hpp

```diff
@@ -0,0 +1,171 @@
+#ifndef VEC4_HPP
+#define VEC4_HPP
+
+#include <cstddef>
+#include <ostream>
+#include <cmath>
+#include <xsimd/xsimd.hpp>
+#include <type_traits>
+
+#include "config.hpp"
+
+/*  *Macros provided by config.hpp
+    - ALIGN_WIDTH: Alignment width for class, since it is dependent on Arch.
+    - UArch: Architecture to be used.
+    - UType: Data type to be used.
+    - CONSTEXPR_CMATH: Switch constexpr if available in cmath.
+*/
+class alignas(ALIGN_WIDTH) Vec4{
+    
+    public:
+        constexpr Vec4() : m_v{0,0,0,0} {}
+        constexpr explicit Vec4(Utype x) : m_v{x,x,x,x} {}
+        constexpr Vec4(Utype x,Utype y,Utype z,Utype w) : m_v{x,y,z,w} {}
+        constexpr explicit Vec4(xsimd::batch<Utype,UArch> x): m_v{0,0,0,0} {x.store_aligned(m_v);}
```

Comment:
If you are assigning to `m_v` inside the body anyways, then the `m_v{0, 0, 0, 0}` part is useless. I would suggest removing it.

Reply:
If i don't do this, i get a compile error.

Reply:
You cannot mark that constructor as `constexpr` because it uses `xsimd::batch` under the hood which is cannot be used in a compile-time context.